### PR TITLE
feat: create a download user data button

### LINKED ISSUE #1713: As a user, I can download my personal information from Chapter
Per https://gdpr-info.eu/art-15-gdpr/ a user has the right to access any of their personal information that we store.  Precisely what constitutes personal information is a bit subjective, so we should err on the side of inclusion.  

With that in mind I think we want data that is connected to their entry in the the `users` table.  Aside from the entry itself we'd want: the details of chapters they are in, their role in these chapters, the same for events, their ban status, and finally their instance_roles. I think that covers it, but let me know if I've missed something.

On the UX side, all we need is a button on their profile to download the data.  The data can be a JSON string.

TODO:
- [x] create a resolver that returns the user's data: https://github.com/freeCodeCamp/chapter/pull/1756
- [ ] create a button to fetch and download that data

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #1713

### CODE REVIEW

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 1]
```diff
@@ -94,9 +94,34 @@ export const UserProfilePage = () => {
...
+              borderRadius={'5px'}
+              paddingBlock={'.65em'}
+              paddingInline={'.4em'}
+              _hover={{ color: 'gray.85', backgroundColor: 'gray.10' }}
+              download={`${userInfo.name}.json`}
+              href={`data:text/json;charset=utf-8,${encodeURIComponent(
+                JSON.stringify(userInfo),
+              )}`}
+            >
+              Download your data
```
Comment (Sboonny) [inline client/src/modules/profiles/pages/userProfile.tsx:122]: interesting, although `area` has `download` and `href` attributes, and should be doing the same thing, it won't be rendered in React. Don't know why tho

File: client/src/modules/profiles/graphql/queries.ts [hunk 2]
```diff
@@ -11,9 +11,11 @@ export const userProfileQuery = gql`
...
         name
       }
-      admined_chapters {
-        id
-        name
+      user_chapters {
+        chapter {
+          id
+          name
+        }
```
Comment (ojeytonwilliams) [inline client/src/modules/profiles/graphql/queries.ts:18]: We should provide them with event and ban info, too. It's data we store about them, after all.

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 3]
```diff
@@ -68,15 +68,15 @@ export const UserProfilePage = () => {
...
-                You are an administrator for these Chapters:
+                You are an member of these Chapters:
               </Heading>
               <Flex marginTop={'1em'} flexDirection={'column'} gap={4}>
-                {userInfo.admined_chapters.map(({ name, id }) => (
-                  <Link key={id} href={`/chapters/${id}`}>
-                    {name}
+                {userInfo.user_chapters.map(({ chapter }) => (
+                  <Link key={chapter.id} href={`/chapters/${chapter.id}`}>
+                    {chapter.name}
```
Comment (ojeytonwilliams) [inline client/src/modules/profiles/pages/userProfile.tsx:79]: Why was this changed?  It's not connected to the download.

File: client/src/modules/profiles/pages/userProfile.tsx (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/modules/profiles/pages/userProfile.tsx:122]: `area` needs to be used with `map`. Why `area`, though?

File: server/src/graphql-types/EventUser.ts [hunk 4]
```diff
@@ -40,3 +40,18 @@ export class EventUser {
...
+
+  @Field(() => Rsvp)
+  rsvp: Rsvp;
+
+  @Field(() => EventRole)
+  event_role: EventRole;
+
+  @Field(() => Event)
+  event: Event;
+}
```
Comment (ojeytonwilliams) [inline server/src/graphql-types/EventUser.ts:57]: I don't think it's necessary to create a new class. We can add `event` to `EventUser` and get it that way.

File: server/src/graphql-types/ChapterUser.ts [hunk 5]
```diff
@@ -25,3 +26,24 @@ export class ChapterUser {
...
+
+  @Field(() => ChapterRole)
+  chapter_role: ChapterRole;
+
+  @Field(() => Chapter)
+  chapter: Chapter;
+
+  @Field(() => Boolean, { nullable: true })
+  is_bannable?: boolean;
+}
```
Comment (ojeytonwilliams) [inline server/src/graphql-types/ChapterUser.ts:49]: Similarly to `UserEvents`, we don't need a new class.

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 6]
```diff
@@ -94,9 +94,34 @@ export const UserProfilePage = () => {
...
+              height={'100%'}
+              size={'lg'}
+              borderRadius={'5px'}
+              paddingBlock={'.65em'}
+              paddingInline={'.4em'}
+              _hover={{ color: 'gray.85', backgroundColor: 'gray.10' }}
+              download={`${userInfo.name}.json`}
+              href={`data:text/json;charset=utf-8,${encodeURIComponent(
+                JSON.stringify(userInfo),
+              )}`}
```
Comment (ojeytonwilliams) [inline client/src/modules/profiles/pages/userProfile.tsx:120]: Since `userInfo` has noise like `__typename` in it, we'd be better off filtering that out.  We can do that via

```ts
const createDownloadData = (userInfo: UserProfileQuery['userInformation']) => {
  const dataString = JSON.stringify(userInfo, (key, value) =>
    key === '__typename' ? undefined : value,
  );
  return `data:text/json;charset=utf-8,${encodeURIComponent(dataString)}`;
};
```

which can go outside the component, since it's a pure function.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: client/src/modules/profiles/pages/userProfile.tsx (hunk 3 above)
Comment (Sboonny) [reply client/src/modules/profiles/pages/userProfile.tsx:79]: Because, the data is the one change as it isn't related to chapters that admin is administrating, but chapters that member has joined

File: server/src/graphql-types/EventUser.ts (hunk 4 above)
Comment (Sboonny) [reply server/src/graphql-types/EventUser.ts:57]: I am worried, about the chance that the class be used somewhere it shouldn't, and we query data it shouldn't. this can be solved with tests, but lets limit the access in queries that use these class, to limit future errors, instead of relaying on memory and tests.

File: client/src/modules/profiles/graphql/queries.ts (hunk 2 above)
Comment (Sboonny) [reply client/src/modules/profiles/graphql/queries.ts]: Added what I think user is interested in 👍

File: server/src/graphql-types/ChapterUser.ts (hunk 5 above)
Comment (ojeytonwilliams) [reply server/src/graphql-types/ChapterUser.ts]: Okay, looking at the hoops you've had to jump through to avoid creating a new class, I take it all back.  Sorry, @Sboonny!

I microscopic change like this (expanding the data type fetched by a single query resolver) should not force you to change other query resolvers.  If they needed the extra data, that would be fine, but they do not.

Here's what I think: we go with

```ts
@ObjectType()
class UserEvent extends EventUserWithRelations {
  @Field(() => Event)
  event: Event;
}
```

and the same for `UserChapter` and use those to make 

```ts
@ObjectType()
export class UserInformation extends UserWithPermissions {
  @Field(() => String)
  email: string;

  @Field(() => [UserBanWithRelations])
  user_bans: UserBanWithRelations[];

  @Field(() => [UserChapter])
  user_chapters: UserChapter[];

  @Field(() => [UserEvent])
  user_events: UserEvent[];
}
```

Then we can create a second query `userDownload`, say, that gets all the download data, but only after the user clicks (so we don't waste time getting it for every visit to the profile).  That means there's no need to change anything else about the profile - the button can go in smoothly.

**Comment by ojeytonwilliams** [comment]
@Sboonny sorry for leading you in the wrong direction.  

Also, we will have to consolidate these object types at some point, but not today!

File: server/src/graphql-types/ChapterUser.ts (hunk 5 above)
Comment (Sboonny) [reply server/src/graphql-types/ChapterUser.ts]: what happens when we extends `UserWithPermissions` that has `user_chapters` and `user_events` types to `UserInformation` that has `user_chapters`, and `user_events` types in it? 

Is the first type declaration ignored?

Comment (ojeytonwilliams) [reply server/src/graphql-types/ChapterUser.ts]: It should overwrite them.  The easiest way to find out is just to try it and see, but the docs are [here](https://typegraphql.com/docs/inheritance.html) if you're interested.

Comment (Sboonny) [reply server/src/graphql-types/ChapterUser.ts]: Ya, I have tested it. just find the idea of overwritten type that you already decided weird, adding and extracting declarations is understandable, but full overwriting it is weirding me out

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 7]
```diff
@@ -94,9 +110,57 @@ export const UserProfilePage = () => {
...
+                      fontWeight="600"
+                      background={'gray.85'}
+                      color={'gray.10'}
+                      height={'100%'}
+                      size={'lg'}
+                      borderRadius={'5px'}
+                      paddingBlock={'.65em'}
+                      paddingInline={'.4em'}
+                      download={`${userData?.name}.json`}
+                      href={createDownloadData(userData)}
```
Comment (Sboonny) [inline client/src/modules/profiles/pages/userProfile.tsx:155]: I have no idea about the logic behind this error in this line, I have checked the owner.json download file and this is the result

```json
{
    "id": 1,
    "name": "The Owner",
    "email": "foo@bar.com",
    "auto_subscribe": false,
    "image_url": null,
    "instance_role": {
        "name": "owner"
    },
    "user_bans": [],
    "user_chapters": [
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 1,
                "name": "Heathcote, Koch and Upton"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 2,
                "name": "Rohan, Franey and Cole"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 3,
                "name": "Cremin Group"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 4,
                "name": "Lynch - Bartoletti"
            }
        }
    ],
    "user_events": []
}
```

I am getting this error.

> Argument of type '{ __typename?: \"UserData\" | undefined; id: number; name: string; email: string; auto_subscribe: boolean; image_url?: string | null | undefined; instance_role: { __typename?: \"InstanceRole\" | undefined; name: string; }; user_bans: { ...; }[]; user_chapters: { ...; }[]; user_events: { ...; }[]; }' is not assignable to parameter of type 'UserProfileQuery'.\n  Types of property '__typename' are incompatible.\n    Type '\"UserData\" | undefined' is not assignable to type '\"Query\" | undefined'.\n      Type '\"UserData\"' is not assignable to type '\"Query\"'.

Comment (Sboonny) [reply client/src/modules/profiles/pages/userProfile.tsx:155]:
```ts
{
    "id": 1,
    "name": "The Owner",
    "email": "foo@bar.com",
    "auto_subscribe": false,
    "image_url": null,
    "admined_chapters": [
        {
            "id": 1,
            "name": "Bradtke - D'Amore",
            "category": "deserunt",
            "city": "Smithamton",
            "country": "Algeria",
            "creator_id": 1,
            "description": "beatae non nostrum",
            "region": "Vermont"
        },
        {
            "id": 2,
            "name": "Orn and Sons",
            "category": "facere",
            "city": "Deckowfield",
            "country": "Martinique",
            "creator_id": 1,
            "description": "neque enim consequatur",
            "region": "Kentucky"
        },
        {
            "id": 3,
            "name": "Rau, Metz and Kulas",
            "category": "aspernatur",
            "city": "Lake Percyton",
            "country": "Cook Islands",
            "creator_id": 1,
            "description": "voluptatem eveniet mollitia",
            "region": "Pennsylvania"
        },
        {
            "id": 4,
            "name": "Kreiger LLC",
            "category": "ut",
            "city": "Port Kennedi",
            "country": "Libyan Arab Jamahiriya",
            "creator_id": 1,
            "description": "a voluptate nisi",
            "region": "Minnesota"
        }
    ],
    "instance_role": {
        "id": 3,
        "name": "owner",
        "instance_role_permissions": [
            {
                "instance_permission": {
                    "name": "chapter-create"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-join"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-subscription-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-user-role-change"
                }
            },
            {
                "instance_permission": {
                    "name": "sponsor-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "sponsor-view"
                }
            },
            {
                "instance_permission": {
                    "name": "user-instance-role-change"
                }
            },
            {
                "instance_permission": {
                    "name": "users-view"
                }
            },
            {
                "instance_permission": {
                    "name": "google-authenticate"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-ban-user"
                }
            },
            {
                "instance_permission": {
                    "name": "event-create"
                }
            },
            {
                "instance_permission": {
                    "name": "event-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "event-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "event-send-invite"
                }
            },
            {
                "instance_permission": {
                    "name": "event-subscription-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp-confirm"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-create"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "chapters-view"
                }
            },
            {
                "instance_permission": {
                    "name": "events-view"
                }
            },
            {
                "instance_permission": {
                    "name": "venues-view"
                }
            }
        ]
    },
    "user_bans": [],
    "user_chapters": [
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 1,
                "name": "Bradtke - D'Amore",
                "description": "beatae non nostrum",
                "category": "deserunt",
                "city": "Smithamton",
                "region": "Vermont",
                "country": "Algeria",
                "creator_id": 1
            },
            "chapter_id": 1,
            "joined_date": "2022-12-01T12:08:51.324Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 2,
                "name": "Orn and Sons",
                "description": "neque enim consequatur",
                "category": "facere",
                "city": "Deckowfield",
                "region": "Kentucky",
                "country": "Martinique",
                "creator_id": 1
            },
            "chapter_id": 2,
            "joined_date": "2022-12-01T12:08:51.326Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 3,
                "name": "Rau, Metz and Kulas",
                "description": "voluptatem eveniet mollitia",
                "category": "aspernatur",
                "city": "Lake Percyton",
                "region": "Pennsylvania",
                "country": "Cook Islands",
                "creator_id": 1
            },
            "chapter_id": 3,
            "joined_date": "2022-12-01T12:08:51.327Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 4,
                "name": "Kreiger LLC",
                "description": "a voluptate nisi",
                "category": "ut",
                "city": "Port Kennedi",
                "region": "Minnesota",
                "country": "Libyan Arab Jamahiriya",
                "creator_id": 1
            },
            "chapter_id": 4,
            "joined_date": "2022-12-01T12:08:51.328Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        }
    ],
    "user_events": []
}
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I went through the userDownload query and found all the things that seemed unnecessary or repetitive.

Also, there shouldn't be a bunch of changes to other query resolvers.  All we want to happen is for the `userData` query resolver to return some stuff.  No other query (or mutation) resolvers should have to change.

Post MVP we can look into making fewer, more generalized resolvers, but for now let's just try to keep each change isolated.

**Comment by Sboonny** [comment]
If we delete that query, the request information will look different from how the UserData is structured, should I change the type for UserData

**Comment by ojeytonwilliams** [comment]
> If we delete that query, the requested information will look different from how the UserData is structured, should I change the type for UserData

Delete which query?

**Comment by Sboonny** [comment]
Delete the data fetching requests in that query*

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 8]
```diff
@@ -94,9 +110,57 @@ export const UserProfilePage = () => {
...
+                      color={'gray.10'}
+                      height={'100%'}
+                      size={'lg'}
+                      borderRadius={'5px'}
+                      paddingBlock={'.65em'}
+                      paddingInline={'.4em'}
+                      download={`${userDataDownload.userData?.name}.json`}
+                      href={createDownloadData(userDataDownload)}
+                    >
+                      Click Here to download the data
```
Comment (ojeytonwilliams) [inline client/src/modules/profiles/pages/userProfile.tsx:157]:
```suggestion
                      Download the data
```

File: client/src/modules/profiles/pages/userProfile.tsx (hunk 3 above)
Comment (ojeytonwilliams) [reply client/src/modules/profiles/pages/userProfile.tsx:95]: Okay, so, we have both `admined_chapters` and `user_chapters` now, so, for this PR it would make most sense to leave this unchanged (since the scope should just be adding the download button).

A separate PR that shows both on the profile page would be good.

File: client/src/modules/profiles/pages/userProfile.tsx [hunk 9]
```diff
@@ -94,9 +110,57 @@ export const UserProfilePage = () => {
...
+            >
+              Request your data
+            </Button>
+            {loading ? (
+              <Spinner />
+            ) : (
+              <>
+                {userDataDownload && (
+                  <Flex alignItems={'center'} gap="2em">
+                    <Text>The data was fetched succesfully 🥳</Text>
```
Comment (ojeytonwilliams) [inline client/src/modules/profiles/pages/userProfile.tsx:144]:
```suggestion
```
I think the link alone is enough.

File: server/src/controllers/Chapter/resolver.ts [hunk 10]
```diff
@@ -68,6 +68,7 @@ export class ChapterResolver {
               },
             },
             user: { include: { instance_role: true } },
+            chapter: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/Chapter/resolver.ts:71]:
```suggestion
```
This should not have changed.

File: server/src/controllers/Chapter/resolver.ts [hunk 11]
```diff
@@ -126,6 +127,7 @@ export class ChapterResolver {
               },
             },
             user: true,
+            chapter: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/Chapter/resolver.ts:130]:
```suggestion
```

File: server/src/controllers/ChapterUser/resolver.ts [hunk 12]
```diff
@@ -25,6 +25,7 @@ import { canBanOther } from '../../util/chapterBans';
 import { updateWaitlistForUserRemoval } from '../../util/waitlist';
 
 const chapterUsersInclude = {
+  chapter: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/ChapterUser/resolver.ts:28]:
```suggestion
```

File: server/src/controllers/EventUser/resolver.ts [hunk 13]
```diff
@@ -38,6 +38,7 @@ export class EventUserResolver {
       },
       where: whereCondition,
       include: {
+        event: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/EventUser/resolver.ts:41]:
```suggestion
```

File: server/src/controllers/EventUser/resolver.ts [hunk 14]
```diff
@@ -72,6 +73,7 @@ export class EventUserResolver {
...
```
Comment (ojeytonwilliams) [inline server/src/controllers/EventUser/resolver.ts:76]:
```suggestion
```

File: server/src/controllers/Events/resolver.ts [hunk 15]
```diff
@@ -62,6 +62,7 @@ import { EventInputs } from './inputs';
 
 const eventUserIncludes = {
   user: true,
+  event: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:65]:
```suggestion
```

File: server/src/controllers/Events/resolver.ts [hunk 16]
```diff
@@ -282,6 +283,7 @@ export class EventResolver {
         event_users: {
           include: {
             user: true,
+            event: true,
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:286]:
```suggestion
```

File: server/src/controllers/Events/resolver.ts [hunk 17]
```diff
@@ -357,6 +359,7 @@ export class EventResolver {
...
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:362]:
```suggestion
```

File: server/src/controllers/Events/resolver.ts [hunk 18]
```diff
@@ -402,6 +405,7 @@ export class EventResolver {
...
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:408]:
```suggestion
```

File: server/src/controllers/Events/resolver.ts [hunk 19]
```diff
@@ -579,7 +583,18 @@ export class EventResolver {
...
+        ...{
+          user: true,
+          rsvp: true,
+          event_role: {
+            include: {
+              event_role_permissions: { include: { event_permission: true } },
+            },
+          },
+        },
+      },
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:597]:
```suggestion
      include: { event: { include: { chapter: true } }, ...eventUserIncludes },
```

File: server/src/controllers/User/resolver.ts [hunk 20]
```diff
@@ -39,6 +40,37 @@ export class UserWithPermissionsResolver {
   async userInformation(
     @Ctx() ctx: ResolverCtx,
   ): Promise<UserInformation | null> {
```
Comment (ojeytonwilliams) [inline server/src/controllers/User/resolver.ts:42]: `UserInformation` is a bit vague, particularly now we also have `UserData`.  I think `UserForDownload` and `UserProfile`, respectively, would be better, so that it's obvious what they're used for.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: client/src/modules/profiles/pages/userProfile.tsx (hunk 9 above)
Comment (Sboonny) [reply client/src/modules/profiles/pages/userProfile.tsx:144]: 👍 I will remove the flex element too, then.

File: server/src/controllers/Chapter/resolver.ts (hunk 10 above)
Comment (Sboonny) [reply server/src/controllers/Chapter/resolver.ts:71]: True, I thought the revert removed them. 👍

**Comment by Sboonny** [comment]
About https://github.com/freeCodeCamp/chapter/pull/1942#discussion_r1040791332, admined_chapters are in `UserDataDownload`, I refactored the types to remove the request for them to save data 😓. Do I revert it?

here is the userProfileQuery.

```ts
export const userProfileQuery = gql`
  query userProfile {
    userInformation {
      id
      name
      email
      auto_subscribe
      image_url
      instance_role {
        name
      }
      admined_chapters {
        id
        name
      }
    }
  }
`;
```

image

**Comment by Sboonny** [comment]
> this is a lie, I don't know how I overwritten the type, and I can't figure it out.

This is a lie, too. Hopefully all sorted with the last commit.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Hey @Sboonny I pushed some (very minor) changes, to save yet another round of reviews.  If you're happy with them, could you merge this?

**Comment by ojeytonwilliams** [comment]
Hey @Sboonny I pushed some (very minor) changes, to save yet another round of reviews.  If you're happy with them, could you merge this?

**Review by Sboonny — COMMENTED** [review]
Couldn't find an issue, thank you for taking care of it 👍
